Task 1/Assets/Scripts/CuttingMeshCalculator.cs

```diff
@@ -0,0 +1,30 @@
+using UnityEngine;
+
+public class CuttingMeshCalculator
+{
+    public struct MeshesData
+    {
+        public Vector3 posStatic;
+        public Vector3 posDynamic;
+        public Vector3 sizeStatic;
+        public Vector3 sizeDynamic;
+    }
+
+    public MeshesData Calculate(Transform prevBlock,Transform curBlock, Vector3 size)
+    {
+        MeshesData data = new MeshesData();
```

Comment:
as curBlock.position is taken 2 times, precache it (don't call C++ side multiple times in one method):
```suggestion
        var curBlockPos = curBlock.position;
        var prevBlockPos = prevBlock.position;
        ...
        reuse this variables
```